File: _CodeTopics/LeetCode_contest/biweekly/biweekly2021/45/45_1.py
class Solution(object):
    def sumOfUnique(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        def list_to_dic(lis):
            dic = {}
            for elem in lis:
                if elem in dic:
                    dic[elem] += 1
                else:
                    dic[elem] = 1
            return dic
        
        dic = list_to_dic(nums)
        res = 0
        for k, v in dic.items():
            if v == 1:
                res += k
        return res
    
"""
https://leetcode-cn.com/contest/biweekly-contest-45/submissions/detail/144244758/ # 被第二道题给气的，不小心关了，发现还有这个链接可以查提交，但是没有详情。

https://leetcode-cn.com/submissions/detail/144244758/

73 / 73 个通过测试用例
状态：通过
执行用时: 24 ms
内存消耗: 13 MB
"""

File: _CodeTopics/LeetCode/1601-1800/001748--biweekly45_1/tran/45_1.cpp
class Solution {
public:
    int sumOfUnique(vector<int>& nums) {

        // 第 45 场双周赛第一题
        // https://leetcode-cn.com/submissions/detail/144244758/

        // 先来个匿名函数都不带引用的试试性能，等会都再加上引用试试。
        std::function<unordered_map<int, int>(vector<int>)> list_to_dic;
        list_to_dic = [](vector<int> lis) -> unordered_map<int, int> {
            unordered_map<int, int> dic{};
            for (auto elem : lis) {
                if (dic.find(elem) != dic.end()) {
                    ++dic[elem];
                } else {
                    dic[elem] = 1;
                }
            }
            return dic;
        };
        
        unordered_map<int, int> dic = list_to_dic(nums);
        int res = 0;
        for (auto [k, v] : dic) {
            if (v == 1) {
                res += k;
            }
        }
        return res;
    }
};

/*
https://leetcode-cn.com/submissions/detail/265066804/

执行用时：0 ms, 在所有 C++ 提交中击败了100.00%的用户
内存消耗：8 MB, 在所有 C++ 提交中击败了5.05%的用户
通过测试用例：
73 / 73
*/
